From d9f21d5208af7c6640b10319094a2a02795a2df0 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Mon, 24 Feb 2020 14:33:42 +0800
Subject: [PATCH] video: rockchip: mpp: fix issue for ioremap the same register
 region

tips: In rk3368, hevc and vdpu use the same register region.
      However, devm_ioremap_resource will call function
      devm_request_mem_region to check region whether map.
      Thus, use devm_ioremap instead to avoid it.

Change-Id: I339b7d5a4e02e9f2304a9c40b146057e1ce13817
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_common.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_common.c b/drivers/video/rockchip/mpp/mpp_common.c
index 6e49c5f8a5a5..2fab291d0b55 100644
--- a/drivers/video/rockchip/mpp/mpp_common.c
+++ b/drivers/video/rockchip/mpp/mpp_common.c
@@ -1378,9 +1378,17 @@ int mpp_dev_probe(struct mpp_dev *mpp,
 		ret = -ENODEV;
 		goto failed;
 	}
-	mpp->reg_base = devm_ioremap_resource(dev, res);
-	if (IS_ERR(mpp->reg_base)) {
-		ret = PTR_ERR(mpp->reg_base);
+	/*
+	 * Tips: here can not use function devm_ioremap_resource. The resion is
+	 * that hevc and vdpu map the same register address region in rk3368.
+	 * However, devm_ioremap_resource will call function
+	 * devm_request_mem_region to check region. Thus, use function
+	 * devm_ioremap can avoid it.
+	 */
+	mpp->reg_base = devm_ioremap(dev, res->start, resource_size(res));
+	if (!mpp->reg_base) {
+		dev_err(dev, "ioremap failed for resource %pR\n", res);
+		ret = -ENOMEM;
 		goto failed;
 	}
 
-- 
2.35.3

